azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/core/DeviceCodeCredentialRetriever.java

```diff
@@ -18,13 +19,14 @@
     private static final String AZURE_TOOLKIT_CLIENT_ID = "777acee8-5286-4d6e-8b05-f7c851d8ed0a";
 
     public DeviceCodeCredentialRetriever(AzureEnvironment env) {
-        this.env = env;
+        super(env);
     }
 
     public AzureCredentialWrapper retrieveInternal() {
+        AuthHelper.setupAzureEnvironment(env);
```

Comment:
Extract set environment to parent class?

Reply:
some time the env is retrieved from auth method, not user's configuration

Reply:
see AzureCliCredentialRetriever